### PR TITLE
Api signin and signout 

### PR DESCRIPTION
**Pull request by macieklad**
Changes:
- API will now authenticate with JWT tokens. Post user credentials to /auth/signin to receive access token. Signout is done by requesting signout route with active user token. More information in api docs.
- Staging env will now be populated with example user to test api requests. It will be presented with each request example in api docs, so look there.

### CODE REVIEW

File: .github/workflows/api.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ env:
   LOG_CHANNEL: stack
   LOG_LEVEL: debug
   DB_CONNECTION: sqlite
+  JWT_SECRET: veCMWIG3Mv6c1w5pzBs4FRp4fBAiYcV5e6rDeiHVW9iZeO4cuYOJeT8CXJ2caQ3f
```
Comment (KalitaKonrad) [inline .github/workflows/api.yml:15]: Shouldn't the secret key be hidden and loaded dynamically e.g. from environment variables?

File: api/app/Http/Traits/RespondsWithJwt.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+   * @param string $token
+   *
+   * @return \Illuminate\Http\JsonResponse
+   */
+  protected function respondWithToken($token)
+  {
+    return response()->json([
+      'access_token' => $token,
+      'token_type' => 'bearer',
+      'expires_in' => Auth::factory()->getTTL() * 60
```
Comment (KalitaKonrad) [inline api/app/Http/Traits/RespondsWithJwt.php:21]: `Auth::factory()->getTTL() * 60` how many minutes is that?

File: api/app/Http/Controllers/Signin.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+     * @bodyParam email string required User email Example: foo@bar.com
+     * @bodyParam password string required User password Example: password
+     * 
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function __invoke(SigninRequest $request)
+    {
+        $credentials = $request->only(['email', 'password']);
+        if (!$token = Auth::attempt($credentials)) {
```
Comment (KalitaKonrad) [inline api/app/Http/Controllers/Signin.php:30]: Shouldn't the equality be done using `==`, not sure tho, not a php expert.

**Review by KalitaKonrad — COMMENTED** [review]
Few questions, other than that LGTM! :) Great job!

File: api/app/Http/Controllers/Signin.php (hunk 3 above)
Comment (macieklad) [reply api/app/Http/Controllers/Signin.php:30]: This is a php feature, conditional assignment, you can set variable inside the if and check that it is truthy at the same time, then reuse it later

File: api/app/Http/Traits/RespondsWithJwt.php (hunk 2 above)
Comment (macieklad) [reply api/app/Http/Traits/RespondsWithJwt.php:21]: By default 60, you can change it in config file, ttl is set in seconds, in this manner - 3600

File: .github/workflows/api.yml (hunk 1 above)
Comment (macieklad) [reply .github/workflows/api.yml:15]: Not here, it is not secret key for external provider that needs to be protected. It is simply used as part of hashing process, so as long as you set it to other value in staging or production env, this key is useless for attacker.

**Pull request merged**